model.js

```diff
@@ -15,12 +16,20 @@ function getArticleById(articleId) {
       return rows[0];
     });
 }
-function getArticles() {
-  return db
-    .query("SELECT * FROM articles ORDER BY created_at DESC")
-    .then(({ rows }) => {
-      return rows;
-    });
+function getArticles(topic) {
+  let query = "SELECT * FROM articles";
+  const topics = [];
+  if (topic) {
+    query += " WHERE topic = $1";
+    topics.push(topic);
+  }
+  query += " ORDER BY created_at DESC";
+  return db.query(query, topics).then(({ rows }) => {
+    if (rows.length === 0) {
+      throw { status: 404, msg: "not found" };
+    }
+    return rows;
```

Comment:
This will need to change for the extra test. You may need to do a separate check topic exists query.